sotodlib/io/imprinter.py

```diff
@@ -277,6 +276,96 @@ def register_books(self, bids, obs_lists, commit=True, session=None):
+            ctime_dirs = sorted(glob(op.join(data_root, "hk", "*")))
+            for ctime_dir in ctime_dirs[:-1]:
+                ctime = op.basename(ctime_dir)
+                book_id = f"hk_{ctime}_{daq_node}"
+                # check whether the book exists
+                if self.get_book(book_id) is not None:
+                    continue
+                book = Books(
+                    bid=book_id,
+                    type="hk",
+                    status=UNBOUND,
+                    tel_tube=daq_node,  # fixme: tel_tube and daq_node may not be the same
+                )
+                session.add(book)
+            if commit: session.commit()
+
+    def _get_binder_for_book(self, book, output_root="out", pbar=False):
+        """get the appropriate bookbinder for the book based on its type"""
+        g3tsmurf_cfg_file = self.sources[book.tel_tube]['g3tsmurf']
+        with open(g3tsmurf_cfg_file, 'r') as f:
+            g3tsmurf_cfg = yaml.safe_load(f)
+        data_root = g3tsmurf_cfg['data_prefix']
+
+        if book.type in ['obs', 'oper']:
+            session_id = book.bid.split('_')[1]
+            first5 = session_id[:5]
+            odir = op.join(output_root, book.type, first5)
+            if not op.exists(odir):
+                os.makedirs(odir)
+            book_path = os.path.join(odir, book.bid)
+
+            # after sanity checks, now we proceed to bind the book.
+            # get files associated with this book, in the form of
+            # a dictionary of {stream_id: [file_paths]}
+            filedb = self.get_files_for_book(book)
+            obsdb = self.get_g3tsmurf_obs_for_book(book)
+            readout_ids = self.get_readout_ids_for_book(book)
+
+            # bind book using bookbinder library
+            bookbinder = BookBinder(book, obsdb, filedb, data_root, readout_ids,
+                                    book_path)
+            return bookbinder
+        
+        elif book.type in ['hk']:
+            # get source directory for hk book
+            hk_root = op.join(data_root, "hk")
+            first5 = book.bid.split('_')[1]
+            assert first5.isdigit(), f"first5 of {book.bid} is not a digit"
+            book_path_src = op.join(hk_root, first5)
+
+            # get target directory for hk book
+            odir = op.join(output_root, book.type)
+            if not op.exists(odir):
+                os.makedirs(odir)
+            book_path_tgt = os.path.join(odir, book.bid)
+
+            class _HKBinder:  # dummy class to mimic baseline bookbinder
+                def __init__(self, indir, outdir):
+                    self.indir = indir
+                    self.outdir = outdir
+                def get_metadata(self):
+                    return {
+                        'book_id': book.bid,
+                        # dummy start and stop times
+                        'start_time': float(first5+"00000"),
+                        'stop_time': float(first5+"99999"),
```

Comment:
I agree with Matthew on the [start, stop) intervals. Those are much nicer to work with.